src/lib/components/utils/acm-tagfield.svelte

```diff
@@ -0,0 +1,129 @@
+<script lang="ts">
+  import { createEventDispatcher } from 'svelte';
+
+  export let tags: string[] = [];
+  export let selectedTags: string[] = [];
+  export let label = '';
+  export let resetButton = '';
+  export let urlSearchParamKey = '';
+
+  const dispatch = createEventDispatcher();
+  let hasSelectedTags = selectedTags.length > 0;
+
+  function selectTag(event: MouseEvent) {
+    event.preventDefault();
+
+    if (selectedTags.includes(this.innerText)) {
+      selectedTags = selectedTags.filter((t) => t !== this.innerText);
+    } else {
+      selectedTags.push(this.innerText);
+    }
+
+    this.classList.toggle('selected');
+    hasSelectedTags = selectedTags.length > 0;
+    dispatch('change', selectedTags);
+  }
+
+  function deselectAll(event: MouseEvent) {
+    event.preventDefault();
+    selectedTags = [];
+    hasSelectedTags = false;
+    dispatch('change', selectedTags);
+  }
+
+  function createTagURL(tag: string) {
+    if (urlSearchParamKey) {
+      if (selectedTags.includes(tag)) {
+        return `?${urlSearchParamKey}=${selectedTags.filter((t) => t !== tag).join(',')}`;
+      } else {
+        return `?${urlSearchParamKey}=${[...selectedTags, tag].join(',')}`;
+      }
+    } else {
+      return '';
+    }
```

Comment:
## Suggestions for `createTagURL()`

### 1) Nit: To avoid bugs, let's make the condition in the if-statement more strict.

```ts
if (urlSearchParamKey === '') {
```

### 2) To keep from making too many indents, let's short circuit.

```ts
if (urlSearchParamKey === '') {
  return '';
}
```

### 3) Let's determine the `nextTags` to be selected before constructing the string.

```ts
const nextTags = selectedTags.includes(tag)
      ? selectedTags.filter((t) => t !== tag)
      : selectedTags.concat([tag]);
```

### 4) Let's put it all together and construct the string.

```suggestion
    if (urlSearchParamKey === '') {
      return '';
    }
    const nextTags = selectedTags.includes(tag)
      ? selectedTags.filter((t) => t !== tag)
      : selectedTags.concat([tag]);
    const params = new URLSearchParams([[urlSearchParamKey, nextTags.join(',')]]);
    return '?' + params.toString();
```